contract/src/agoric-basics.contract.js

```diff
@@ -40,58 +40,112 @@ const bagCounts = bag => {
   const entries = getCopyBagEntries(bag);
   return entries.map(([_k, ct]) => ct);
 };
+
+/**
+ *
+ * @param {import('@endo/patterns').CopyBag} bag
+ * @param {Object.<string, {tradePrice: Amount, maxTickets: bigint}>} inventory
+ * @returns {boolean}
+ */
+export const hasInventory = (bag, inventory) => {
+  const entries = getCopyBagEntries(bag);
+  for (const [k, ct] of entries) {
+    if (k in inventory) {
+      const { maxTickets } = inventory[k];
+      if (ct > maxTickets) {
+        return false;
+      }
+    } else {
+      return false;
+    }
+  }
+
+  return true;
+};
+
+/**
+ *
+ * @param {Amount} amount
+ * @param {number} n
+ * @returns {Amount}
+ */
+const multiply = (amount, n) => {
```

Comment:
_question_

I didn't see a `AmountMath.multiply` function so I implemented it myself

Reply:
I seem to have some trouble making `multiply` work for `Amount<AssetKind>` instead of `Amount<'nat'>`, am I missing something?

Reply:
Our ratio library doesn't support non-fungible amounts and says so in its type annotations... which means you may have to trace thru all the callers and update their types to `Amount<'nat'>`.

In some places, a typecast is called for. For example, `zcf.getTerms()` doesn't know whether the brands are for fungible assets or not. And you're not in a position to doubt the caller that started the contract - that's part of the [reliance set](https://papers.agoric.com/assets/pdf/papers/concurrency-among-strangers.pdf). You could ask the brand what its AssetKind is, but it could lie, so it's not worth the bother.